tactical-microgrid-standard/controller/CLIClient.cpp

```diff
@@ -0,0 +1,608 @@
+#include "CLIClient.h"
+#include "qos/QosHelper.h"
+
+#include <dds/DCPS/PublisherImpl.h>
+#include <dds/DCPS/SubscriberImpl.h>
+#include <dds/DCPS/Marked_Default_Qos.h>
+#include <dds/DCPS/WaitSet.h>
+
+#include <cctype>
+#include <thread>
+
+CLIClient::CLIClient(const tms::Identity& id)
+  : handshaking_(id)
+  , stop_cli_(false)
+{}
+
+DDS::ReturnCode_t CLIClient::init(DDS::DomainId_t domain_id, int argc, char* argv[])
+{
+  DDS::ReturnCode_t rc = handshaking_.join_domain(domain_id, argc, argv);
+  if (rc != DDS::RETCODE_OK) {
+    return rc;
+  }
+
+  // Subscribe to the DeviceInfo and Heartbeat to learn about microgrid controllers
+  // But not publish to these topics.
+  rc = handshaking_.create_subscribers(
+    [&](const tms::DeviceInfo& di, const DDS::SampleInfo& si) { process_device_info(di, si); },
+    [&](const tms::Heartbeat& hb, const DDS::SampleInfo& si) { process_heartbeat(hb, si); });
+  if (rc != DDS::RETCODE_OK) {
+    return rc;
+  }
+
+  DDS::DomainParticipant_var dp = handshaking_.get_domain_participant();
+
+  // Publish to the PowerDevicesRequest topic
+  cli::PowerDevicesRequestTypeSupport_var pdreq_ts = new cli::PowerDevicesRequestTypeSupportImpl;
+  if (DDS::RETCODE_OK != pdreq_ts->register_type(dp.in(), "")) {
```

Comment:
```suggestion
  if (DDS::RETCODE_OK != pdreq_ts->register_type(dp, "")) {
```
I typically take these out